sdk/generic-sdk-core/generic-core/src/test/java/util/ConfigurationSource.java

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package util;
+
+import java.util.Map;
+
+/**
+ * Configuration property source which provides configuration values from a specific place. Samples may include
+ * properties file supported by frameworks or other source.
+ *
+ * Note that environment configuration (environment variables and system properties) are supported by default and
+ * don't need a source implementation.
+ */
+@FunctionalInterface
+public interface ConfigurationSource {
```

Comment:
Should this not be a test class and be in `generic-core` instead? Is it used outside of our current tests?